File: backend/src/main/java/com/exercise/backend/service/TaskService.java
package com.exercise.backend.service;
import com.exercise.backend.model.Task;
import com.exercise.backend.model.TaskState;
import com.exercise.backend.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Optional;

@Service
public class TaskService {

    private TaskRepository taskRepository;

    @Autowired
    public TaskService(TaskRepository taskRepository) {
        this.taskRepository = taskRepository;
    }

    public ResponseEntity<Task> createTask(Task task) {
        Optional<Task> optionalTask = Optional.of(task);
        if (optionalTask.isPresent()) {
            task.setState(TaskState.UNCOMPLETED);
            taskRepository.save(task);
        }
        return ResponseEntity.of(optionalTask);
    }

    public Collection<Task> getTasks() {
        Collection<Task> tasks = new ArrayList<>();
        taskRepository.findAllOrderById().forEach(task -> tasks.add(task));
        return tasks;
    }

    public ResponseEntity<Task> updateTask(Long taskId, Task task) {
        Optional<Task> optionalTask = taskRepository.findById(taskId);
        if (optionalTask.isPresent()) {
            task.setTaskId(taskId);
            taskRepository.save(task);
        }
        return ResponseEntity.of(optionalTask);
    }

    public ResponseEntity<Object> deleteTask(Long taskId) {
        if (taskRepository.existsById(taskId)) {
            taskRepository.deleteById(taskId);
            return ResponseEntity.ok().build();
        }
        return ResponseEntity.notFound().build();
    }
}

File: backend/src/main/frontend/src/service/TaskService.js
import axios from "axios";

class TaskService {

    async fetchTasks() {
        const response = await axios.get("http://localhost:8080/tasks");
        return response;
    }

    async updateTask(task) {
        const response = await axios.put(`http://localhost:8080/tasks/update/${task.taskId}`, task);
        return response;
    }

    async createTask(task) {
        const response = await axios.post("http://localhost:8080/tasks/create/", task);
        return response;
    }

    async deleteTask(task) {
        const response = await axios.delete(`http://localhost:8080/tasks/delete/${task.taskId}`);
        return response;
    }
};

export default TaskService;
